metrics/metrics.go

```diff
@@ -1,84 +1,63 @@
+// Copyright (C) 2020 NTT Innovation Institute, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package metrics
 
 import (
 	"fmt"
-	"net"
-	"strconv"
 	"time"
 
 	l "github.com/cloudwan/gohan/log"
 	"github.com/cloudwan/gohan/util"
-	graphite "github.com/cyberdelia/go-metrics-graphite"
-	"github.com/rcrowley/go-metrics"
+	metrics "github.com/rcrowley/go-metrics"
 )
 
 var (
-	graphiteConfigs   []graphite.Config
+	exporter          metricsExporter
```

Comment:
why global variable?

Reply:
The public interface for metrics is free functions (e.g. `StartMetricsProcess()`) so they have to operate on some kind of singleton. I have not changed the public interface at all, notice that previously `graphiteConfigs` were global - I've simply abstracted that using an interface.

Reply:
I see, can we group those global variables to some struct, or is it too much work for now?